linalg/hypre.cpp

```diff
@@ -2336,6 +2365,20 @@ HyprePCG::~HyprePCG()
 }
 
 
+HypreGMRES::HypreGMRES(MPI_Comm comm) : precond(NULL)
+{
+   int k_dim    = 50;
+   int max_iter = 100;
+   double tol   = 1e-6;
```

Comment:
Since these values are used in multiple places, I think it is best to make them `static const` members of the class with names like `k_dim_default`, `max_iter_default`, etc.

Reply:
Good point.  Perhaps a better choice is to write a private member function `SetDefaultOptions` which can be called by both constructors.

Reply:
Yes, that is also a good option.